bcipy/helpers/raw_data.py

```diff
@@ -58,22 +65,84 @@ def channels(self) -> List[str]:
     @property
     def numeric_data(self) -> pd.DataFrame:
         """Data for columns with numeric data. This is usually comprised of the
-        timestamp column and device channels, excluding triggers."""
+        timestamp column and device channels, excluding string triggers."""
         return self.dataframe.select_dtypes(exclude=['object'])
 
-    def by_channel(self) -> np.ndarray:
+    def by_channel(self, transform: Optional[Composition] = None) -> np.ndarray:
         """Data organized by channel.
 
+        Optionally, it can apply a BciPy Composition to the data before returning using the transform arg.
+        This will apply n tranformations to the data before returning. For an example Composition with
+        EEG preprocessing, see bcipy.signal.get_default_transform().
+
         Returns
         ----------
-        C x N numpy array with samples where C is the number of channels and N
-        is number of time samples."""
+        data: C x N numpy array with samples where C is the number of channels and N
+        is number of time samples
+        fs: resulting sample rate if any transformations applied"""
 
         numeric_data = self.numeric_data
 
         numeric_vals = numeric_data.values
         numeric_column_count = numeric_vals.shape[1]
-        return numeric_vals[:, 1:numeric_column_count].transpose()
+        data = numeric_vals[:, 1:numeric_column_count].transpose()
+        fs = self.sample_rate
+
+        if transform:
+            data, fs = self.apply_transform(data, transform)
+
+        return data, fs
+
+    def to_mne(
```

Comment:
What do you think about moving this functionality to the `convert` module?

Reply:
My initial reasoning when we made the ticket was this would be the first of several MNE PRs to support better data processing/handling across the codebase- that we would start relying on their formats more heavily, deprecating some of the BciPy redundancies where we could. It made sense when I thought of it in that way to support going in-between RawData formats here. It does add some additional overhead to the RawData module. I'm open to rethinking how we handle data to support ease of use and quality data processing.

Reply:
I think it makes sense to leverage more of what MNE provides. The raw_data module could be a lot more efficient by  handling the underlying data as typed arrays (using pandas, numpy, or MNE RawArray) rather than reading everything as lists. However, in other areas of the codebase we provide our own interfaces and avoid committing too heavily on a single library (GUI module -> PyQt; Displays -> psychopy; acquisition -> LSL; etc), which helps us maintain a consistent API in case any of these things change.

We are using a separate module for data conversion functions, which I think is a good idea to separate concerns. It seems consistent to treat MNE as another data format that goes here. Instead of calling `raw_data.to_mne()` you might call `convert.to_mne(raw_data)` or `convert_to_mne(raw_data)`.

We might also think about whether conversion is necessary or if additional behavior should be provided by RawData (that could dispatch to MNE).

Reply:
> We are using a separate module for data conversion functions, which I think is a good idea to separate concerns. It seems consistent to treat MNE as another data format that goes here. Instead of calling raw_data.to_mne() you might call convert.to_mne(raw_data) or convert_to_mne(raw_data).

That seems reasonable to me while we are discussing MNE integration and future multimodal needs. I'll make the change!

> We might also think about whether conversion is necessary or if additional behavior should be provided by RawData (that could dispatch to MNE).

Another consideration is that we may start using devices not ideal or yet integrated (tripolars) for MNE use... we will likely have to abstract several things, but ultimately we need better support for common BCI data needs (exports by varying dimensions & data type, filtering out channels by map or type, setting of sensor coordinates).